File: dotnet/Models/EmailMessage.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace AvailabilityNotify.Models
{
    public class JsonData
    {
        public GetSkuContextResponse SkuContext { get; set; }
        public NotifyRequest NotifyRequest { get; set; }
    }

    public class EmailMessage
    {
        public object ProviderName { get; set; }
        public string TemplateName { get; set; }
        public JsonData JsonData { get; set; }
    }
}

File: node/typings/emailMessage.ts
import { GetSkuContextResponse } from './getSkuContext'
import { NotifyRequest } from './notifyRequest'

export interface JsonData {
  SkuContext: GetSkuContextResponse
  NotifyRequest: NotifyRequest
}

export interface EmailMessage {
  ProviderName: unknown
  TemplateName: string
  JsonData: JsonData
}
